doc/new_boilerplate_migration.md

```diff
@@ -0,0 +1,635 @@
+# New OverReact Boilerplate Migration Guide
+
+* __[Background](#background)__
+    * [Function Component Boilerplate](#function-component-boilerplate)
+    * [Problems with Previous Boilerplate](#problems-with-previous-boilerplate)
+* __[New Boilerplate Updates](#new-boilerplate-updates)__
+    * [Constraints](#constraints)
+    * [Updates](#updates)
+        * [Remove Annotations](#remove-annotations)
+        * [Ignore Ungenerated Warnings Project-Wide](#ignore-ungenerated-warnings-project-wide)
+        * [Use Mixin-Based Props Declaration that Disallows Subclassing](#use-mixin-based-props-declaration-that-disallows-subclassing)
+* __[Function Components](#function-components)__
+    * [Constraints](#function-component-constraints)
+    * [Design](#design)
+* __[Upgrading](#upgrading)__
+
+## Background
+
+While converting the boilerplate for OverReact component declaration from Dart 1 
+transformers to Dart 2 builders, we encountered several constraints that made us 
+choose between dropping backwards compatibility (mainly support for props class 
+inheritance), and a less-than-optimal boilerplate.
+
+To make the Dart 2 transition as smooth as possible, we chose to keep the new 
+boilerplate version as backwards-compatible as possible, while compromising the 
+cleanliness of the boilerplate. In time, we found that this wasn't great from a 
+user experience or from a tooling perspective.
+
+Knowing this, we now have the opportunity to implement an improved version of 
+OverReact boilerplate that fixes issues introduced in the latest version, as 
+well as other miscellaneous ones.
+
+### Function Component Boilerplate
+
+React function components, along with [hooks](https://reactjs.org/docs/hooks-intro.html), 
+are used heavily in modern React component architecture, and we would like to 
+take advantage of their benefits within the Workiva ecosystem. However, to be 
+truly valuable, we need strongly typed prop APIs for these components, which 
+means we must provide a way to declare function components using OverReact 
+boilerplate.
```

Comment:
Since this isn't part of the "migration", I'm not sure this belongs here.  If we want to keep it here - it should go in some sort of section talking about "new boilerplate features" or something like that.

Reply:
```suggestion
### Function Component Boilerplate

React function components, along with [hooks](https://reactjs.org/docs/hooks-intro.html), 
are used heavily in modern React component architecture, and we want consumers to 
be able to take advantage of their benefits. However, to be 
truly valuable, we need strongly typed prop APIs for these components, which 
means we must provide a way to declare function components using OverReact 
boilerplate.
```